src/dbnode/storage/index/block.go

```diff
@@ -217,23 +193,32 @@ type BlockOptions struct {
 func NewBlock(
 	blockStart time.Time,
 	md namespace.Metadata,
-	opts BlockOptions,
-	indexOpts Options,
+	blockOpts BlockOptions,
+	opts Options,
 ) (Block, error) {
 	blockSize := md.Options().IndexOptions().BlockSize()
-	iopts := indexOpts.InstrumentOptions()
+	iopts := opts.InstrumentOptions()
+	scope := iopts.MetricsScope().SubScope("index").SubScope("block")
+	mutableSegments := newMutableSegments(
+		blockStart,
+		opts,
+		blockOpts,
+		iopts,
+		scope,
```

Comment:
Instead of passing both iopts and scope here, maybe just reset iopts and pass that in?

```go
iopts := opts.InstrumentOptions()
scope := iopts.MetricsScope().SubScope("index").SubScope("block")
iopts = iopts.SetMetricsScope(scope)
```